chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -0,0 +1,644 @@
+
+				chainntnfs.Log.Infof("New block: height=%v, sha=%v",
+					item.Height, item.Hash)
+
+				b.notifyBlockEpochs(item.Height, &item.Hash)
+
+				txns := btcutil.NewBlock(rawBlock).Transactions()
+				err = b.txConfNotifier.ConnectTip(&item.Hash,
+					uint32(item.Height), txns)
+				if err != nil {
+					chainntnfs.Log.Error(err)
+				}
+				continue
+
+			case chain.BlockDisconnected:
+				if item.Height != currentHeight {
+					chainntnfs.Log.Warnf("Received blocks "+
+						"out of order: current height="+
+						"%d, disconnected height=%d",
+						currentHeight, item.Height)
+					continue
+				}
+
+				currentHeight = item.Height - 1
+
+				chainntnfs.Log.Infof("Block disconnected from "+
+					"main chain: height=%v, sha=%v",
+					item.Height, item.Hash)
+
+				err := b.txConfNotifier.DisconnectTip(
+					uint32(item.Height))
+				if err != nil {
+					chainntnfs.Log.Error(err)
+				}
+
+			case chain.RelevantTx:
+				tx := item.TxRecord.MsgTx
+				// First, check if this transaction spends an output
+				// that has an existing spend notification for it.
+				for i, txIn := range tx.TxIn {
+					prevOut := txIn.PreviousOutPoint
+
+					// If this transaction indeed does spend an
+					// output which we have a registered
+					// notification for, then create a spend
+					// summary, finally sending off the details to
+					// the notification subscriber.
+					if clients, ok := b.spendNotifications[prevOut]; ok {
+						spenderSha := tx.TxHash()
+						spendDetails := &chainntnfs.SpendDetail{
+							SpentOutPoint:     &prevOut,
+							SpenderTxHash:     &spenderSha,
+							SpendingTx:        &tx,
+							SpenderInputIndex: uint32(i),
+						}
+						// TODO(roasbeef): after change to
+						// loadfilter, only notify on block
+						// inclusion?
+						if item.Block != nil {
+							spendDetails.SpendingHeight = item.Block.Height
+						} else {
+							spendDetails.SpendingHeight = currentHeight + 1
+						}
+
+						for _, ntfn := range clients {
+							chainntnfs.Log.Infof("Dispatching "+
+								"spend notification for "+
+								"outpoint=%v", ntfn.targetOutpoint)
+							ntfn.spendChan <- spendDetails
+
+							// Close spendChan to ensure that any calls to Cancel will not
+							// block. This is safe to do since the channel is buffered, and the
+							// message can still be read by the receiver.
+							close(ntfn.spendChan)
+						}
+						delete(b.spendNotifications, prevOut)
+					}
+				}
+			}
+
+		case <-b.quit:
+			break out
+		}
+	}
+	b.wg.Done()
+}
+
+// historicalConfDetails looks up whether a transaction is already included in a
+// block in the active chain and, if so, returns details about the confirmation.
+func (b *BitcoindNotifier) historicalConfDetails(txid *chainhash.Hash,
+) (*chainntnfs.TxConfirmation, error) {
+
+	// If the transaction already has some or all of the confirmations,
+	// then we may be able to dispatch it immediately.
+	tx, err := b.chainConn.GetRawTransactionVerbose(txid)
```

Comment:
Having a workaround for when `getrawtransaction` can't be used (`txindex=0`) would be nice. @Roasbeef I assume that was part of your plan; I'm still trying to connect some dots :-)

Reply:
Yes, but that'll be a follow up after this PR gets in.